src/various_small_datasets/gen_api/views.py

```diff
@@ -213,19 +242,17 @@ def get_queryset(self):
         return self.model.objects.all()
 
     def get_serializer_class(self):
-        if self.dataset not in GenericViewSet.serializerClasses:
-            GenericViewSet.serializerClasses[self.dataset] = serializers.serializer_factory(self.dataset, self.model)
-        return GenericViewSet.serializerClasses[self.dataset]
+        as_geojson = False
+        if 'as_geojson' in self.request.query_params:
+            as_geojson = True
+
+        key = f'{self.dataset}_{as_geojson}'
+
+        if key not in GenericViewSet.serializerClasses:
+            GenericViewSet.serializerClasses[key] = serializers.serializer_factory(self.dataset, self.model, as_geojson)
+        return GenericViewSet.serializerClasses[key]
```

Comment:
Hier moeten we  verschillende GenericViewSet.serializerClasses voor geojson en normale serializers hebben. Anders  worden geojson en niet geojson door elkaar heen gemixt, afhankelijk van wat het eerst wordt  gebruikt